index.js

```diff
@@ -1,47 +1,163 @@
 'use strict';
-const util = require('util');
-const fs = require('fs');
-const chalk = require('chalk');
+
 const Jimp = require('@sindresorhus/jimp');
-const termImg = require('term-img');
+const chalk = require('chalk');
+const {string: termImg} = require('term-img');
+const terminalCharWidth = require('terminal-char-width');
+
+const CHAR_HEIGHT = 16;
+const CHAR_WIDTH = CHAR_HEIGHT * terminalCharWidth;
+
+/**
+ * Check if two colors are equal
+ *
+ * Both has `alpha=0`, or `alpha>0` and equl all the other color values
+ *
+ * @returns {Boolean} colors are equal
+ */
+function equalColor({r, g, b, a}, {r: r2, g: g2, b: b2, a: a2}) {
+	if (a === 0 && a2 === 0) {
+		return true;
+	}
+
+	return a > 0 && a2 > 0 && r === r2 && g === g2 && b === b2;
+}
+
+function getBgColor({r, g, b, a}) {
+	// Full transparent
+	if (a === 0) {
+		return chalk.reset;
+	}
+
+	return chalk.bgRgb(r, g, b);
+}
+
+function getFgColor({r, g, b}) {
+	return chalk.rgb(r, g, b);
+}
+
+function getGlyph(upper, lower) {
+	if (upper.a) {
+		if (lower && equalColor(upper, lower)) {
+			return '█';
```

Comment:
Is it really worth all this complication just to save some glyphs? Like, what are the benefits of using a whole glyph in this scenario?

Reply:
Both the upper and lower half blocks has the same color, so it can be defined with a full block glyph and the foreground color and the background don't need to change. This complication allow use less ANSI escape codes. There's the problem `chalk` add reset ANSI codes at the end that are mostly useless since we are going the set the color again later, so another change I was thinking is to use directly the `ansi-styles` module to do the color transitions, but this can be done in a different pull-request.

Reply:
👍 Alright